### DISCUSSION TITLE
Output Voltage Gauge - 0V

### DISCUSSION THREAD

**Discussion by FileCity** [nutdotnet/WinNUT-Client#188] [Ideas]
My old APC Back-UPS RS 1500 I just rebuild doesn't have the output voltage variable.
To avoid having this gauge at 0V, i think it may be a good idea to just follow the input voltage as a cosmetic measure.
When AC is present, input and output will follow each other and on battery (with outage), both will be at 0V.
Some UPSes may have a different output voltage when doing TRIM or BOOST and, for them, should be reported as is.
As a complement of information, here are the available variables of my UPS:

apc1500 (American Power Conversion/Back-UPS RS 1500/8.g8 .D)
battery.charge (Battery charge (percent of full)) : 100
battery.charge.low (Remaining battery level when UPS switches to LB (percent)) : 33
battery.charge.warning (Battery level when UPS switches to Warning state (percent)) : 50
battery.date (Battery change date) : 2001/09/25
battery.mfr.date (Battery manufacturing date) : 2021/11/21
battery.runtime (Battery runtime (seconds)) : 15688
battery.runtime.low (Remaining battery runtime when UPS switches to LB (seconds)) : 600
battery.type (Battery chemistry) : PbAc
battery.voltage (Battery voltage (V)) : 26.6
battery.voltage.nominal (Nominal battery voltage (V)) : 24.0
device.mfr (Description unavailable) : American Power Conversion
device.model (Description unavailable) : Back-UPS RS 1500
device.serial (Description unavailable) : ------------ REMOVED
device.type (Description unavailable) : ups
driver.flag.ignorelb (Description unavailable) : enabled
driver.name (Driver name) : usbhid-ups
driver.parameter.lowbatt (Description unavailable) : 33
driver.parameter.offdelay (Description unavailable) : 90
driver.parameter.ondelay (Description unavailable) : 120
driver.parameter.pollfreq (Description unavailable) : 30
driver.parameter.pollinterval (Description unavailable) : 3
driver.parameter.port (Description unavailable) : auto
driver.parameter.synchronous (Description unavailable) : no
driver.version (Driver version - NUT release) : 2.7.4
driver.version.data (Description unavailable) : APC HID 0.96
driver.version.internal (Internal driver version) : 0.41
input.sensitivity (Input power sensitivity) : high
input.transfer.high (High voltage transfer point (V)) : 126
input.transfer.low (Low voltage transfer point (V)) : 104
input.transfer.reason (Reason for last transfer to battery) : input voltage out of range
input.voltage (Input voltage (V)) : 118.0
input.voltage.nominal (Nominal input voltage (V)) : 120
ups.beeper.status (UPS beeper status) : disabled
ups.delay.shutdown (Interval to wait after shutdown with delay command (seconds)) : 90
ups.firmware (UPS firmware) : 8.g8 .D
ups.firmware.aux (Auxiliary device firmware) : g8
ups.load (Load on UPS (percent of full)) : 12
ups.mfr (UPS manufacturer) : American Power Conversion
ups.mfr.date (UPS manufacturing date) : 2005/01/27
ups.model (UPS model) : Back-UPS RS 1500
ups.productid (Product ID for USB devices) : 0002
ups.realpower.nominal (UPS real power rating (W)) : 865
ups.serial (UPS serial number) : ------------ REMOVED
ups.status (UPS status) : OL
ups.test.result (Results of last self test) : No test initiated
ups.timer.reboot (Time before the load will be rebooted (seconds)) : 0
ups.timer.shutdown (Time before the load will be shutdown (seconds)) : -1
ups.vendorid (Vendor ID for USB devices) : 051d

**Comment by gbakeman** [comment]
Hey, great to hear from you again! Hope you've been well.

Here's a few other discussions sort of relating to this:

- https://github.com/nutdotnet/WinNUT-Client/issues/136
- https://github.com/nutdotnet/WinNUT-Client/issues/116

I see where you're coming from with the idea to tie the non-existent output voltage gauge to the input voltage gauge, as there's not really any reason for them to be different on AC anyways. I'm just a little hesitant to be presenting a made-up or assumed reading to the user. The functionality of WinNUT right now is to give a default value when one does not exist on the server, and up until recently the default values were typically nominal (non-NA grid ones if I recall correctly.) I recently made a change to start giving obvious out of range defaults (0 or -1) to some values at least, and this has caught a few people by surprise because they thought the fake nominal readings were real.

I'm still wrestling with how to present out of range or non-existent values on the gauges themselves, but in the bigger picture I think I'd like to restructure WinNUT for multiple UPSs as well as allowing the user greater customization over what gauges are displayed (so missing variables won't be displayed at all.) Does this sound like a good plan to you, or do you have other thoughts?

**Comment by Grumpy-Old-Engineer** [comment]
I've got similar/same problem with a CyberPower UPS.  Is there anything the casual user can do to correct UPS data displayed? I've got a good UPS, and the UPS front panel display shows values I'm expecting, but the Output Voltage and Battery Voltage shown on the WinNUT Client display just aren't right.
![WinNUT_Display](https://github.com/user-attachments/assets/0d9c732d-82eb-4694-b666-c79af124956a)

**Reply by gbakeman** [reply]
Hi there,

Your issue does seem slightly different from @FileCity's, although we do get questions about CyberPower UPSs here from time to time. I'd like to ask that you [open a new discussion](https://github.com/nutdotnet/WinNUT-Client/discussions/new?category=q-a) with the information you provided above, along with the variable output from File -> UPS Variables and an upload of the WinNUT log file set to debug mode. Thanks in advance!

**Comment by FileCity** [comment]
@gbakeman, for the output voltage issue, another approach could be to grey-out this gauge, put it out of focus, etc, and remove the needle.
So the visual aspect of the application will be kept intact and the user will know this gauge will not function.
Having it activated with a 0V output can be confusing for some.
What do you think ?

**Reply by gbakeman** [reply]
That sounds great to me. I'd recommend following the two issues above where I'll aim to fix this issue completely. Please feel free to add your thoughts there as well. I'm balancing this with using a new build & deployment system so it may take me a little bit.
